test.js

```diff
@@ -89,4 +89,29 @@ describe('markdown-it-implicit-figures', function() {
     var res = md.render(src);
     assert.equal(res, expected);
   });
+
+  it('should put the image inside a link to the image if it is not yet linked', function () {
+    md = Md().use(implicitFigures, { link: true });
+    var src = '![www.google.com](fig.png)';
+    var expected = '<figure><a href="fig.png"><img src="fig.png" alt="www.google.com"></a></figure>\n';
```

Comment:
I like this because it puts the `<a>` as a child of `<figure>` (and the parent of `<img>`, obviously). It's fully semantic and doesn't change the previous output style (or mess with `<figcaption>`ing 👍).